ffac-update-location-gps/luasrc/usr/bin/update-location-gps

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/lua
+
+if not arg[1] then
+    print("Error: No /dev/tty* path specified. Please provide a TTY device path as the first argument.")
+    os.exit(1)
+end
+
+local uci = require('simple-uci').cursor()
+local bit = require 'bit'
+local fcntl = require 'posix.fcntl'
+local unistd = require 'posix.unistd'
+
+-- Get GPS device tty
+local tty_device = arg[1]
+
+-- Use GPS as Stream
+local file = io.open(tty_device, "r")
+if not file then
+    print("Error: Unable to open " .. tty_device)
+    os.exit(2)
+end
+
+local lockfilename = "/var/lock/update-location-gps_" .. string.gsub(tty_device, "/", "_")
+
+local lockfd, err = fcntl.open(lockfilename, bit.bor(fcntl.O_WRONLY, fcntl.O_CREAT), 384) -- mode 0600
+if not lockfd then
+  print('err', err)
```

Comment:
I am not sure about this print format here. Above you are using `print("Error: Unable to open " .. tty_device)` and below `print("GPS position is Invalid...", valid)`.

Some kind of uniform error or success printing would be nice to make it easier debug issues where this is deployed.